### PR TITLE
expect_forces_chunked statistics minor fix

### PR DESCRIPTION
**Pull request by Quantumgunhee**
Based on the issue #2010, I fixed the transpose operation in the expect_forces_chunked statistics.
I verified the correct statistics from the previously reported code.

### CODE REVIEW

**Review by gcarleo — APPROVED** [review]
I would merge this as it is, since it seems like an annoying bug for some

**Comment by PhilipVinc** [comment]
@Quantumgunhee could you also fix the test that should have caught this bug?

The test case is [this one](https://github.com/netket/netket/blob/a2e87926d42d50c9815996a1613bd535ca797d37/test/variational/test_variational.py#L514) , and as you can see we are checking that the gradients computed with and without chunk size match
```python
    vstate.chunk_size = None
    grad_nochunk = vstate.grad(operator)
    vstate.chunk_size = chunk_size
    grad_chunk = vstate.grad(operator)

    jax.tree_util.tree_map(
        partial(np.testing.assert_allclose, atol=1e-13), grad_nochunk, grad_chunk
    )

```

However, instead of checking the output of `vstate.grad` we should have been checking the output of `vstate.expect_and_grad`. By simply changing this, we can make sure this does not happen again.

**Comment by ltdavidson77** [comment]
of 1e-RDM or S2 value, I need an accuracy of 1e-4, which is not too strict. Then I used 8 A100 cards for parralization training and inferencing on a fermi Hubbard model with 20 sites. The training took about 40 minutes to converge to the required accuracy, while the inference of the 1e-RDM and S2 took more than 6 hours to reach an uncertainty of 2e-4, which is computed as the $\sqrt{\sigma^2/N}$ . I feel depressed by this, and, if this is the case, other computations like the phase diagram would be very hard to carry out.

Do you have any suggestions for this?

Thanks!

Zhanghao
